packages/sdk-btc/source/transaction.service.ts

```diff
@@ -1,55 +1,265 @@
+			// 1. Derive the sender address (corresponding to first address index for the wallet)
+			const { address, type, path } = await this.addressService.fromMnemonic(
+				input.signatory.signingKey(),
+				identityOptions,
+			);
+			console.log(type, address, path);
 
-			// 1. Derive the sender address
-			const { address } = await this.addressService.fromWIF(input.signatory.signingKey());
-			// ({ wif: input.signatory.signingKey() });
+			// 3. Compute the amount to be transferred
+			const amount = this.toSatoshi(input.data.amount).toNumber();
 
-			// 2. Aggregate the unspent transactions
-			const unspent: UnspentTransaction[] = await this.#unspent.aggregate(address);
+			// 4. Add utxos
+			const accountKey = BIP32.fromMnemonic(input.signatory.signingKey(), network)
+				.deriveHardened(levels.purpose)
+				.deriveHardened(levels.coinType)
+				.deriveHardened(levels.account || 0);
 
-			// 3. Compute the amount to be transfered
-			const amount = this.toSatoshi(input.data.amount).toNumber();
+			console.log("accountKey", accountKey.toBase58(), accountKey.neutered().toBase58());
 
-			// 4. Build and sign the transaction
-			let transaction = new Transaction().from(unspent).to(input.data.to, amount).change(address);
+			const targets = [
+				{
+					address: input.data.to,
+					value: amount,
+				},
+			];
 
-			// 5. Set a fee if configured. If none is set the fee will be estimated by bitcore-lib.
-			if (input.fee) {
-				const fee = this.toSatoshi(input.fee).toNumber();
-				transaction = transaction.fee(fee);
-			}
+			const feeRate = await this.#getFee(input);
+
+			const { inputs, outputs, fee } = await this.#selectUtxos(levels, accountKey, targets, feeRate);
+
+			const psbt = new bitcoin.Psbt({ network: network });
 
-			return transaction.sign(input.signatory.signingKey()).toString();
+			inputs.forEach((input) => {
+				return psbt.addInput({
+					hash: input.txId,
+					index: input.vout,
+					...input,
+				});
+			});
+			outputs.forEach((output) => {
+				// watch out, outputs may have been added that you need to provide
+				// an output address/script for change
+				if (!output.address) {
+					output.address = address; // @TODO Derive and use fresh change addresses wallet.getChangeAddress()
+					// wallet.nextChangeAddress()
+				}
+
+				psbt.addOutput({
+					address: output.address,
+					value: output.value,
+				});
+			});
+
+			inputs.forEach((input, index) => psbt.signInput(index, bitcoin.ECPair.fromPrivateKey(input.signingKey)));
+
+			psbt.validateSignaturesOfAllInputs();
+			psbt.finalizeAllInputs();
+
+			const transaction: bitcoin.Transaction = psbt.extractTransaction();
+
+			return this.dataTransferObjectService.signedTransaction(
+				transaction.getId(),
+				{
+					sender: address,
+					recipient: input.data.to,
+					amount: amount,
+					fee,
+					timestamp: new Date(), // TODO See if we have a proper timestamp inside the built transaction
```

Comment:
@faustbrian Txs don't have a timestamp. Do we leave it like this?
https://bitcoin.stackexchange.com/questions/77538/how-to-get-time-of-input-and-output-of-a-given-a-transaction

Reply:
That should be fine. I have to do this for some other coins too because they have no timestamps.